mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -0,0 +1,99 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.james.backends.cassandra.utils.CassandraUtils;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.table.CassandraAnnotationTable;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.store.mail.AnnotationMapper;
+import org.apache.james.mailbox.store.transaction.NonTransactionalMapper;
+
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.Select;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+public class CassandraAnnotationMapper extends NonTransactionalMapper implements AnnotationMapper {
+
+    private final CassandraId mailboxId;
+    private final Session session;
+
+    public CassandraAnnotationMapper(CassandraId mailboxId, Session session) {
+        this.mailboxId = mailboxId;
+        this.session = session;
+    }
+
+    public List<MailboxAnnotation> getAllAnnotations() {
+        return CassandraUtils.convertToStream(session.execute(getStoredAnnotationsQuery()))
+            .map(this::toAnnotation)
+            .collect(Collectors.toList());
+    }
+
+    public List<MailboxAnnotation> getAnnotationsByKeys(Set<String> keys) {
+        return CassandraUtils.convertToStream(session.execute(getStoredAnnotationsQueryForKeys(keys)))
+            .map(this::toAnnotation)
+            .collect(Collectors.toList());
+    }
+
+    public void deleteAnnotation(String key) {
+        session.execute(delete().from(CassandraAnnotationTable.TABLE_NAME)
+            .where(eq(CassandraAnnotationTable.MAILBOX_ID, mailboxId.asUuid()))
+            .and(eq(CassandraAnnotationTable.KEY, key)));
+    }
+
+    public void insertAnnotation(MailboxAnnotation mailboxAnnotation) {
+        Preconditions.checkArgument(!mailboxAnnotation.isNil());
+        session.execute(insertInto(CassandraAnnotationTable.TABLE_NAME)
+            .value(CassandraAnnotationTable.MAILBOX_ID, mailboxId.asUuid())
+            .value(CassandraAnnotationTable.KEY, mailboxAnnotation.getKey())
+            .value(CassandraAnnotationTable.VALUE, mailboxAnnotation.getValue().get()));
+    }
+
+    private MailboxAnnotation toAnnotation(Row row) {
+        return MailboxAnnotation.newInstance(row.getString(CassandraAnnotationTable.KEY), row.getString(CassandraAnnotationTable.VALUE));
+    }
+
+    private Select.Where getStoredAnnotationsQuery() {
+        return select(CassandraAnnotationTable.SELECT_FIELDS)
+                .from(CassandraAnnotationTable.TABLE_NAME)
+                .where(eq(CassandraAnnotationTable.MAILBOX_ID, mailboxId.asUuid()));
+    }
+
+    private Select.Where getStoredAnnotationsQueryForKeys(Set<String> keys) {
+        return getStoredAnnotationsQuery().and(in(CassandraAnnotationTable.KEY, Lists.newArrayList(keys)));
```

Comment:
I don't understand where you think the order is meaningful, could you show me the code ?